### PR TITLE
Update Unreal Engine options

### PR DESCRIPTION
**Pull request by dciborow**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Update Unreal Engine 4 from 4.27 to 4.27.2
* Update Unreal Engine 5 from 5.0.EA to 5.0.1

### CODE REVIEW

**Comment by dciborow** [comment]
@bmoore-msft , looks like I need this VM image approved on the test subscription.
Offer details: publisher='microsoftcorporation1602274591143' offer = 'game-dev-vm', sku = 'win10_unreal_4_27_2'

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 1]
```diff
@@ -134,18 +137,12 @@ param publicIpAllocationMethod string = 'Dynamic'
 ])
 param publicIpSku string = 'Basic'
 
-@description('Public IP With or None?')
-@allowed([
-  'new'
-  'existing'
-  'none'
-])
+@description('Public IP New or Existing or None?')
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:140]:
```suggestion
@description('Public IP With or None?')
@allowed([
  'new'
  'existing'
  'none'
])
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 2]
```diff
@@ -134,18 +137,12 @@ param publicIpAllocationMethod string = 'Dynamic'
...
 param publicIpNewOrExisting string = 'new'
 
 @description('Resource Group of the Public IP Address')
 param publicIpRGName string = resourceGroup().name
 
-@description('Select Image Deployment for debugging only')
 @allowed([
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:146]:
```suggestion
@description('Select Image Deployment for debugging only')
@allowed([
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 3]
```diff
@@ -209,26 +216,71 @@ var environments = {
 var vmImage = environments[environment].vmImage
 var vmPlan = environments[environment].vmPlan
 
-var ipconfName = '${vmName}-ipconf'
-var nicName = '${vmName}-nic'
-var nsgName = '${vmName}-nsg'
+var vmName_var = vmName
+var ipconfName = '${vmName_var}-ipconf'
+var nicName_var = '${vmName_var}-nic'
+var nsgName_var = '${vmName_var}-nsg'
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:222]:
```suggestion
var ipconfName = '${vmName}-ipconf'
var nicName = '${vmName}-nic'
var nsgName = '${vmName}-nsg'
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 4]
```diff
@@ -209,26 +216,71 @@ var environments = {
...
 
-var storageType = (bool(length(split(vmSize, '_')) > 2) ? 'Premium_LRS' : 'Standard_LRS')
+var storageType_var = (bool(length(split(vmSize, '_')) > 2) ? 'Premium_LRS' : 'Standard_LRS')
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:224]:
```suggestion
var storageType = (bool(length(split(vmSize, '_')) > 2) ? 'Premium_LRS' : 'Standard_LRS')
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 5]
```diff
@@ -209,26 +216,71 @@ var environments = {
...
 
-var tags = {
+var tags_var = {
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:226]:
```suggestion
var tags = {
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 6]
```diff
@@ -240,12 +292,12 @@ resource partnercenter 'Microsoft.Resources/deployments@2021-04-01' = {
   }
 }
 
-resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicIpNewOrExisting == 'new') {
+resource publicIp 'Microsoft.Network/publicIPAddresses@2021-03-01' = if (publicIpNewOrExisting == 'new') {
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:295]:
```suggestion
resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicIpNewOrExisting == 'new') {
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 7]
```diff
@@ -240,12 +292,12 @@ resource partnercenter 'Microsoft.Resources/deployments@2021-04-01' = {
...
   name: publicIpName
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:296]:
```suggestion
  name: publicIpName
  location: location
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 8]
```diff
@@ -240,12 +292,12 @@ resource partnercenter 'Microsoft.Resources/deployments@2021-04-01' = {
...
-  location: location
   sku: {
     name: publicIpSku
   }
+  location: location
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:300]:
```suggestion
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 9]
```diff
@@ -254,246 +306,23 @@ resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicI
...
-        }
-      ]
-    }['nsgRules-${remoteAccessTechnology}']
+    securityRules: nsg_rules.outputs.nsgRules['nsgRules-${remoteAccessTechnology}']
   }
 }
 
-resource vnet 'Microsoft.Network/virtualNetworks@2021-05-01' = if (vnetNewOrExisting == 'new') {
+
+resource vnet 'Microsoft.Network/virtualNetworks@2021-03-01' = if (vnetNewOrExisting == 'new') {
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:325]:
```suggestion
resource vnet 'Microsoft.Network/virtualNetworks@2021-05-01' = if (vnetNewOrExisting == 'new') {
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 10]
```diff
@@ -254,246 +306,23 @@ resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicI
...
-            direction: 'Inbound'
-            sourceAddressPrefix: '*'
-            sourcePortRange: '*'
-            destinationAddressPrefix: '*'
-            destinationPortRange: '80'
-          }
-        }
-      ]
-    }['nsgRules-${remoteAccessTechnology}']
+    securityRules: nsg_rules.outputs.nsgRules['nsgRules-${remoteAccessTechnology}']
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:320]:
```suggestion
    securityRules: {
      'nsgRules-RDP': !unrealPixelStreamingEnabled ? [
        {
          name: 'RDP'
          properties: {
            priority: 1010
            protocol: '*'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '3389'
          }
        }
      ] : [
        {
          name: 'RDP'
          properties: {
            priority: 1010
            protocol: '*'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '3389'
          }
        }
        {
          name: 'PixelStream'
          properties: {
            priority: 1020
            protocol: '*'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '80'
          }
        }
      ]
      'nsgRules-Teradici': !unrealPixelStreamingEnabled ? [
        {
          name: 'RDP'
          properties: {
            priority: 1010
            protocol: '*'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '3389'
          }
        }
        {
          name: 'PCoIPtcp'
          properties: {
            priority: 1020
            protocol: 'TCP'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '4172'
          }
        }
        {
          name: 'PCoIPudp'
          properties: {
            priority: 1030
            protocol: 'UDP'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '4172'
          }
        }
        {
          name: 'CertAuthHTTPS'
          properties: {
            priority: 1040
            protocol: 'TCP'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '443'
          }
        }
        {
          name: 'TeradiciCom'
          properties: {
            priority: 1050
            protocol: 'TCP'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '60443'
          }
        }
      ] : [
        {
          name: 'RDP'
          properties: {
            priority: 1010
            protocol: '*'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '3389'
          }
        }
        {
          name: 'PCoIPtcp'
          properties: {
            priority: 1020
            protocol: 'TCP'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '4172'
          }
        }
        {
          name: 'PCoIPudp'
          properties: {
            priority: 1030
            protocol: 'UDP'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '4172'
          }
        }
        {
          name: 'CertAuthHTTPS'
          properties: {
            priority: 1040
            protocol: 'TCP'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '443'
          }
        }
        {
          name: 'TeradiciCom'
          properties: {
            priority: 1050
            protocol: 'TCP'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '60443'
          }
        }
        {
          name: 'PixelStream'
          properties: {
            priority: 1060
            protocol: '*'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '80'
          }
        }
      ]
      'nsgRules-Parsec': !unrealPixelStreamingEnabled ? [
        {
          name: 'RDP'
          properties: {
            priority: 1010
            protocol: '*'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '3389'
          }
        }
      ] : [
        {
          name: 'RDP'
          properties: {
            priority: 1010
            protocol: '*'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '3389'
          }
        }
        {
          name: 'PixelStream'
          properties: {
            priority: 1020
            protocol: '*'
            access: 'Allow'
            direction: 'Inbound'
            sourceAddressPrefix: '*'
            sourcePortRange: '*'
            destinationAddressPrefix: '*'
            destinationPortRange: '80'
          }
        }
      ]
    }['nsgRules-${remoteAccessTechnology}']
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 11]
```diff
@@ -258,7 +309,7 @@ resource nsg 'Microsoft.Network/networkSecurityGroups@2021-05-01' = {
   name: nsgName
   location: location
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:310]:
```suggestion
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 12]
```diff
@@ -256,9 +306,8 @@ resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicI
 
 resource nsg 'Microsoft.Network/networkSecurityGroups@2021-05-01' = {
   name: nsgName
-  location: location
   properties: {
-    securityRules: {
+        securityRules: {
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:310]:
```suggestion
    securityRules: {
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 13]
```diff
@@ -493,6 +542,7 @@ resource nsg 'Microsoft.Network/networkSecurityGroups@2021-05-01' = {
   }
 }
 
+
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:545]:
```suggestion
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 14]
```diff
@@ -541,14 +591,13 @@ resource nic 'Microsoft.Network/networkInterfaces@2021-05-01' = {
   }
 }
 
-resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-11-01' = {
+resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-03-01' = {
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:594]:
```suggestion
resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-11-01' = {
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 15]
```diff
@@ -566,16 +615,23 @@ resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-11-01' = {
...
+      windowsConfiguration: {
+        enableAutomaticUpdates: (bool(windowsUpdateOption != 'Manual') ? true : false)
+        patchSettings: {
+          patchMode: windowsUpdateOption
+        }
+      }
+      customData: base64(customData)
     }
-    userData: base64(userData)
+    // userData: base64(userData)
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:634]:
```suggestion
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 16]
```diff
@@ -584,35 +640,42 @@ resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-11-01' = {
...
+    }
   }
 }
 
-resource virtualMachine_enableAAD 'Microsoft.Compute/virtualMachines/extensions@2021-11-01' = if (enableAAD) {
-  name: '${virtualMachine.name}/AADLoginForWindows'
-  location: location
-  dependsOn: [
-    remoteAccess
+resource virtualMachine_enableAAD 'Microsoft.Compute/virtualMachines/extensions@2019-12-01' = if(enableAAD) {
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:674]:
```suggestion
resource virtualMachine_enableAAD 'Microsoft.Compute/virtualMachines/extensions@2021-11-01' = if (enableAAD) {
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 17]
```diff
@@ -584,35 +640,42 @@ resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-11-01' = {
...
+  name      : '${virtualMachine.name}/AADLoginForWindows'
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:675]:
```suggestion
  name: '${virtualMachine.name}/AADLoginForWindows'
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 18]
```diff
@@ -584,35 +640,42 @@ resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-11-01' = {
...
+  location  : location
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:676]:
```suggestion
  location: location
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 19]
```diff
@@ -584,35 +640,42 @@ resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-11-01' = {
...
+  dependsOn : [
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:677]:
```suggestion
  dependsOn: [
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 20]
```diff
@@ -222,13 +235,51 @@ var tags = {
...
+var subnetId = '${vnetId[vnetNewOrExisting]}/subnets/${subNetName}'
+
+var publicIpId = {
+  'new': resourceId('Microsoft.Network/publicIPAddresses', publicIpName)
+  'existing': resourceId(publicIpRGName, 'Microsoft.Network/publicIPAddresses', publicIpName) 
+  'none': ''
+}[publicIpNewOrExisting]
 
-resource partnercenter 'Microsoft.Resources/deployments@2021-04-01' = {
+resource partnercenter 'Microsoft.Resources/deployments@2020-06-01' = {
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:282]:
```suggestion
resource partnercenter 'Microsoft.Resources/deployments@2021-04-01' = {
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 21]
```diff
@@ -242,7 +293,6 @@ resource partnercenter 'Microsoft.Resources/deployments@2021-04-01' = {
 
 resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicIpNewOrExisting == 'new') {
   name: publicIpName
-  location: location
   sku: {
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:296]:
```suggestion
  location: location,
  sku: {
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 22]
```diff
@@ -242,7 +293,7 @@ resource partnercenter 'Microsoft.Resources/deployments@2021-04-01' = {
 
 resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicIpNewOrExisting == 'new') {
   name: publicIpName
-  location: location
+  location: location,
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:296]:
```suggestion
  location: location
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 23]
```diff
@@ -256,7 +307,6 @@ resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicI
 
 resource nsg 'Microsoft.Network/networkSecurityGroups@2021-05-01' = {
   name: nsgName
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:309]:
```suggestion
  name: nsgName
  location: location
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 24]
```diff
@@ -584,35 +639,42 @@ resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-11-01' = {
       ]
     }
   }
+  tags: (contains(outTagsByResource, 'Microsoft.Compute/virtualMachines') ? union(tags, outTagsByResource['Microsoft.Compute/virtualMachines']) : tags)
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:642]:
```suggestion
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 25]
```diff
@@ -222,11 +235,49 @@ var tags = {
...
+parsecHost={16}
+parsecUserEmail={17}
+parsecIsGuestAccess={18}
+
+deployedFromSolutionTemplate={19}
+''', fileShareStorageAccount, fileShareStorageAccountKey, fileShareName, p4Port, p4Username, p4Password, p4Workspace, p4Stream, p4ClientViews, ibLicenseKey, gdkVersion, useVmToSysprepCustomImage, remoteAccessTechnology, teradiciRegKey, parsec_teamId, parsec_teamKey, parsec_host, parsec_userEmail, parsec_isGuestAccess, deployedFromSolutionTemplate)
+
+var vnetId = {
+  'new'     : resourceId('Microsoft.Network/virtualNetworks', vnetName)
+  'existing': resourceId(vnetRGName, 'Microsoft.Network/virtualNetworks', vnetName)
```
Comment (bmoore-msft) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:272]: this mapping isn't needed, the syntax you have for ```existing``` will work in all cases

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 26]
```diff
@@ -222,11 +235,49 @@ var tags = {
...
+}
+var subnetId = '${vnetId[vnetNewOrExisting]}/subnets/${subNetName}'
+
+var publicIpId = {
+  'new': resourceId('Microsoft.Network/publicIPAddresses', publicIpName)
+  'existing': resourceId(publicIpRGName, 'Microsoft.Network/publicIPAddresses', publicIpName)
```
Comment (bmoore-msft) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:278]: this mapping isn't needed, the syntax you have for ```existing``` will work in all cases

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 27]
```diff
@@ -586,33 +643,39 @@ resource virtualMachine 'Microsoft.Compute/virtualMachines@2021-11-01' = {
...
-    fileShareStorageAccount: fileShareStorageAccount
-    fileShareStorageAccountKey: fileShareStorageAccountKey
-    fileShareName: fileShareName
-    p4Port: p4Port
-    p4Username: p4Username
-    p4Password: p4Password
-    p4Workspace: p4Workspace
-    p4Stream: p4Stream
-    p4ClientViews: p4ClientViews
+resource virtualMachine_GDVMCustomization 'Microsoft.Compute/virtualMachines/extensions@2019-12-01' = if (deployedFromSolutionTemplate) {
```
Comment (bmoore-msft) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:646]: why is this only done when deployed from marketplace?

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 28]
```diff
@@ -621,3 +684,7 @@ resource virtualMachine_enableAAD 'Microsoft.Compute/virtualMachines/extensions@
     autoUpgradeMinorVersion: true
   }
 }
+
+output Host_Name string = (!empty(publicIpId) ? reference(publicIpId, '2021-03-01').dnsSettings.fqdn : '')
+output UserName string = adminName
+output IPAddress string = (!empty(publicIpId) ? publicIpId : '')
```
Comment (bmoore-msft) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:690]: this isn't an ipAddress it's a resourceId - what's the intent?

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep (hunk 28 above)
Comment (dciborow) [reply application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep]: should have returned ip address, I hope this is fixed via "publicIp.ipAddress"

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 29]
```diff
@@ -621,3 +673,7 @@ resource virtualMachine_enableAAD 'Microsoft.Compute/virtualMachines/extensions@
     autoUpgradeMinorVersion: true
   }
 }
+
+output Host_Name string = publicIp.dnsSettings.fqdn
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:677]:
```suggestion
output Host_Name string = publicIp.properties.dnsSettings.fqdn
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 30]
```diff
@@ -621,3 +673,7 @@ resource virtualMachine_enableAAD 'Microsoft.Compute/virtualMachines/extensions@
...
+output UserName string = adminName
+output IPAddress string = publicIp.ipAddress
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:679]:
```suggestion
output IPAddress string = publicIp.properties.ipAddress
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep (hunk 27 above)
Comment (dciborow) [reply application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:635]: we have not yet moved these scripts to github, and instead they are now included directly within the VM image. Let me take this back under consideration, and we may decide to also include the files here.

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep [hunk 31]
```diff
@@ -621,3 +673,7 @@ resource virtualMachine_enableAAD 'Microsoft.Compute/virtualMachines/extensions@
     autoUpgradeMinorVersion: true
   }
 }
+
+output Host_Name string = publicIp.properties.dnsSettings.fqdn
+output UserName string = adminName
+output IPAddress string = publicIp.properties.ipAddress
```
Comment (dciborow) [inline application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep:679]:
```suggestion
```

File: application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep (hunk 27 above)
Comment (dciborow) [reply application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep]: would like to take on adding those files in a separate PR and lock in these updates.

**Pull request merged**